backend/config/functions.php

```diff
@@ -26,4 +30,63 @@ function apiReturn($status_code, $response){
 function sanitizeStr($str){
     return htmlentities($str, ENT_QUOTES, "UTF-8");
 }
-?>
\ No newline at end of file
+
+function randomStr($length){
+   return bin2hex(random_bytes($length));
+}
+
+function getCsrf(){
+    if (isset($_SESSION['csrf'])) {
+        $token_age = $timestamp - $_SESSION['csrf_time'];
+        if ($token_age <= $config['site']['csrf_time']){  
+            return $_SESSION['csrf'];
+        } else {
+            $_SESSION['csrf'] = sha1(randomStr().$user_agent);
+            $_SESSION['csrf_time'] = $timestamp;
+            return $_SESSION['csrf'];
+        }
+    } else {
+        $_SESSION['csrf'] = sha1(randomStr().$user_agent);
+        $_SESSION['csrf_time'] = $timestamp;
+        return $_SESSION['csrf'];
+    }
+    setcookie("csrf", $_SESSION['csrf'], $config['site']['csrf_time'], '/', $config['site']['domain'], true, false, 'Lax');
+}
+
+function verifyCsrf($token){
+    if (isset($_SESSION['csrf'])) {
+        $token_age = $timestamp - $_SESSION['csrf_time'];
+        if ($token_age <= $config['site']['csrf_time']){ 
+            if ($token == $_SESSION['csrf']) {
+                    unset($_SESSION['csrf']);
+                    unset($_SESSION['csrf_time']);
+                    return true;
+                } else {
+                unset($_SESSION['csrf']);
+                unset($_SESSION['csrf_time']);
+                return false;
+                }
+            } else {
+                unset($_SESSION['csrf']);
+                unset($_SESSION['csrf_time']);
+                return false;
+            }
```

Comment:
To jest bardzo pogmatwane i można uprościć, na przykład tak:

```suggestion
        $token_age = $timestamp - $_SESSION['csrf_time'];
        $r = ($token_age <= $config['site']['csrf_time'] && $token == $_SESSION['csrf']);
        unset($_SESSION['csrf']);
        unset($_SESSION['csrf_time']);
        return $r;
```

Można nawet bardziej uprościć. Generalnie im mniej kodu tym lepiej, bo:
1. Mam mniej do czytania na review
2. Jak się wraca do tego to jest mniej do czytania
3. Mniej kodu = mniej miejsc gdzie może się coś zepsuć